### PR TITLE
Document Concur Request v4 cash advance & exception endpoints

### PR DESCRIPTION
**Pull request by laurentlhermine**
+ Various enhancements and corrections

### CODE REVIEW

File: src/api-reference/request/v4.endpoints.request-resources.markdown [hunk 1]
```diff
@@ -208,10 +215,12 @@ GET {datacenter}/travelrequest/v4/requests
...
...the approved Requests by a user (relevant only for the approver). Closed Requests are included in this view;<br>`CANCELED`: Get all the cancelled Requests for a user (relevant only for the traveler). Cancelled could include close/not closed requests;<br>`CLOSED`: Get all the closed Requests for a user (relevant only for the traveler). Includes canceled then closed Request as well as approved then closed Requests;<br>`SUBMITTED`: Get all the submitted Requests for a user (relevant only for the traveler). Submitted does not include cancelled requests;<br>`TOAPPROVE`: Get all Requests to be approved by the user (relevant only for the approver);<br>`PENDINGEBOOKING`: Approved Requests awaiting Concur Travel booking(s);<br>`PENDINGPROPOSAL`: Get all Requests submitted to a Travel Agency (TMC) step step. The userId of the TMC agent user must be provided in the list of parameters, this TMC agent user must have a default Travel Agency assigned in its profile corresponding to the Travel Agency assigned to the Requests.<br>When connecting with a User Token, additional values are possible: `PROPOSALAPPROVED`, `PROPOSALCANCELED`.<br>If no view value is sent, the default view `ALL` will be used
```
Comment (sarraloew) [inline src/api-reference/request/v4.endpoints.request-resources.markdown]: A few text edits, also is there a reason to not alphabetize these values? 
"perimeter" should be "parameter"
PENDING: Does pending mean submitted but not yet approved? or not yet final approved (could have some approvals but not all required approvals)? Please clarify.
CANCELED "close/not closed" should be "closed/not closed"
PENDINGPROPOSAL has an extra "step"
PROPOSALAPPROVED and PROPOSALCANCELLED should also have definitions.

File: src/api-reference/request/v4.endpoints.schemas.markdown [hunk 2]
```diff
@@ -41,9 +45,11 @@ Name|Type|Format|Description
 approvalLimitDate|timestamp|[RFC 3339] yyyy-MM-dd'T'HH:mm:ss.SSS'Z'|The date by which the Request must be approved. This element appears only when integrated with Concur Travel
 approvalStatus|object|[Approval Status](#schema-approvalstatus)|The approval status of the Request|
 approved|boolean|-|Indicates whether this Request is approved
+approver|object|[Employee](#schema-employee)|The approver to whom the Request was sent to
```
Comment (sarraloew) [inline src/api-reference/request/v4.endpoints.schemas.markdown]: Remove the final "to"

File: src/api-reference/request/v4.endpoints.schemas.markdown [hunk 3]
```diff
@@ -84,9 +92,29 @@ currency|string|[ISO 4217:2015]|**Required** The 3-letter ISO 4217 code for the
...
+
+Name|Type|Format|Description
+---|---|---|---
+amountRequested|object|[Amount](#schema-amount)|The amount of the cash advance in the Request, expressed in the reimbursement currency of the employee at the time they created the cash advance
+approvalStatus|object|[Cash Advance Approval Status](#schema-cashadvance-approvalstatus)|The approval status of the cash advance
+cashAdvanceId|string|-|The unique identifier of the cash advance
+comment|string|-|The last comment related to this cash advance
+exchangeRate|object|[Exchange Rate](#schema-exchangerate)|The exchange rate that applies to the cash advance
+issueDate|timestamp|[RFC 3339] yyyy-MM-dd'T'HH:mm:ss.SSS|The date the cash advance was issued
+requestDate|timestamp|[RFC 3339] yyyy-MM-dd'T'HH:mm:ss.SSS|The date the cash advance was submitted (last submit date in case of recall))
```
Comment (sarraloew) [inline src/api-reference/request/v4.endpoints.schemas.markdown]: remove extra parentheses

**Review by sarraloew — CHANGES_REQUESTED** [review]
I've got a few text edits and one question. Thanks!

File: src/api-reference/request/v4.endpoints.schemas.markdown [hunk 4]
```diff
@@ -116,6 +144,24 @@ lastName|string|-|The last name of the employee
...
+isBlocking|boolean|-|Defines whether the exception will prevent the Request from being submitted
+level|integer|-|The numeric level associated with the exception. Example: 99
+message|string|-|The user-facing message defined for the exception
+source|array|[Exception source](#schema-exceptionsource)|The source that has raised the exception
+
+### <a name="schema-exceptionsource"></a>Exception Source
+
+Name|Type|Format|Description
+---|---|---|---
+href|string||[RFC 3986]|Hyperlink to the resource
```
Comment (sarraloew) [inline src/api-reference/request/v4.endpoints.schemas.markdown:160]: This line has an extra pipe. Also the swagger doc has additional information about this field that we should have here too: Will be empty if source is ALLOCATION or CASH_ADVANCE.

File: src/api-reference/request/v4.endpoints.schemas.markdown (hunk 2 above)
Comment (sarraloew) [reply src/api-reference/request/v4.endpoints.schemas.markdown]: Also the swagger has this additional information, let's add it here too: Null when Request is in an approval status different from SUBMITTED

File: src/api-reference/request/v4.endpoints.schemas.markdown [hunk 5]
```diff
@@ -52,8 +58,10 @@ custom1 to custom20|object|[CustomField](#schema-customfield)|The details from t
 endDate|date|[ISO 8601] yyyy-MM-dd|The end date of the Request
 endTime|time|[ISO 8601] yyyy-MM-dd|The end time of the Request
 everSentBack|boolean|-|Indicates whether the Request has ever been sent back to the employee
+exceptions|object|[ResourceLink](#schema-resourcelink)|The list of exceptions that have been raised to this Request
 expenses|array|[ResourceLink](#schema-resourcelink)|Expected expenses attached to this Request
 extensionOf|object|[RequestLink](#schema-requestlink)|The Request that this Request is an extension of, or addendum to
+highestExceptionLevel|string|-|The highest level of exception contained in this Request. <br>Possible values: `WARNING`, `ERROR`
```
Comment (sarraloew) [inline src/api-reference/request/v4.endpoints.schemas.markdown]: The swagger file includes NONE as a possible value for highestExceptionLevel- which is correct?

**Comment by quancheng-zhao** [comment]
src/api-reference/request/v4.endpoints.schemas.markdown: in CustomField schema definition, listItemId is an old value and it needs to be removed.

File: src/api-reference/request/v4.endpoints.cashadvance-resources.markdown [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+---
+
+{% include prerelease.html %}
+
+* [Request Cash Advance Resources](#request-cash-advance-resource)
+  * [Get the list of existing cash advances within a Request](#get-request-cash-advance-resource)
+
+# <a name="request-cash-advance-resource"></a>Request Cash Advance Resources
+
+## <a name="get-request-cash-advance-resource"></a>Get the content of an existing cash advances
```
Comment (quancheng-zhao) [inline src/api-reference/request/v4.endpoints.cashadvance-resources.markdown:13]: Get the content of an existing cash advance

**Comment by quancheng-zhao** [comment]
v4.endpoints.request-resources.markdown: GET {datacenter}/travelrequest/v4/requests/{requestUuid}/comments doesn't have userId as parameter

File: src/api-reference/request/v4.endpoints.schemas.markdown [hunk 7]
```diff
@@ -116,6 +143,24 @@ lastName|string|-|The last name of the employee
...
 
+### <a name="schema-exceptions"></a>Exceptions
+
+Name|Type|Format|Description
+---|---|---|---
+code|string|-|The system exception code defined for the exception. Example: BADCODE
+isBlocking|boolean|-|Defines whether the exception will prevent the Request from being submitted
+level|integer|-|The numeric level associated with the exception. Example: 99
+message|string|-|The user-facing message defined for the exception
+source|array|[Exception source](#schema-exceptionsource)|The source that has raised the exception
```
Comment (quancheng-zhao) [inline src/api-reference/request/v4.endpoints.schemas.markdown:154]: Missing an attribute "parameters"? (Parameters for the messages of the exception)

**Comment by quancheng-zhao** [comment]
Expected Expense Resource's description updates to "Manage expected expenses attached to a Request"

**Review by sarraloew — APPROVED** [review]

**Pull request merged**